File: Battle-Jawn/src/main/java/com/battlejawn/Battle/Jawn/EnemyMove/Impale.java
package com.battlejawn.Battle.Jawn.EnemyMove;

import com.battlejawn.Battle.Jawn.Interfaces.CriticalHit;

public class Impale implements CriticalHit {
    private int damage;

    public int getDamage() {
        return damage;
    }

    public void setDamage(int damage) {
        this.damage = damage;
    }

    public void attack() {
        setDamage(/* enemy.strength * 1.2 */ 10);

        if (criticalHit()){
            setDamage(damage *= 1.5);
        }
    }

    public boolean criticalHit() {
        int chance = (int) Math.floor(Math.random() * 100);
        if (chance > 90) {
            return true;
        }
        return false;
    }
}

File: legacy/player/playerAttack/impale.js
import { logBox } from "../../UI/logBox/logBox.js";
import { user } from "../../user/user.js";
import { createEnemy } from "../../enemy/createEnemy.js";
import { userInterface } from "../../UI/UI.js";
import { logBoxDisplay } from "../../UI/logBox/logBoxDisplay.js";
import { enemyMoves } from "../../enemy/enemyMoves.js";

export function impale() {
    let damage = user.strength * 1.2;

    let criticalHitChance = Math.floor(Math.random() * 100)
    
        if (criticalHitChance > 90) {
            damage = Math.floor(damage * 1.5);
            logBox.push("Critical Hit!");
        }

    createEnemy.health = createEnemy.health - damage;
    logBox.push(`You used Impale! You did ${damage} damage.`);
    userInterface();
    logBoxDisplay();
    enemyMoves();
    //During real time combat, will have a 3 second cool down
}